exchange/exchange.go

```diff
@@ -204,7 +204,9 @@ func (e *exchange) HoldAuction(ctx context.Context, r AuctionRequest, debugLog *
 	if err != nil {
 		return nil, err
 	}
-	requestExt.Prebid.Server = &openrtb_ext.ExtRequestPrebidServer{ExternalUrl: e.server.ExternalUrl, GvlID: e.server.GvlID, Datacenter: e.server.Datacenter}
+	if requestExt.Prebid.Server.Empty() {
+		requestExt.Prebid.Server = &openrtb_ext.ExtRequestPrebidServer{ExternalUrl: e.server.ExternalUrl, GvlID: e.server.GvlID, DataCenter: e.server.DataCenter}
+	}
```

Comment:
Ah my earlier thought here wasn't about checking if there was any server data specified in the request, but rather about checking if there was any host data so that we only set `requestExt.Prebid.Server` if `e.server` is non-empty in order to avoid creating an object at `ext.prebid.server` without any real data in it.

My understanding is that server info specified in the host config should always overwrite what is specified in the request.

@SyntaxNode, your suggestion was that it might make sense to put an `Empty` method on `ExtRequestPrebidServer`. Were you thinking the logic would then be something like this?
```
serverInfo := openrtb_ext.ExtRequestPrebidServer{ExternalUrl: e.server.ExternalUrl, GvlID: e.server.GvlID, DataCenter: e.server.DataCenter}
if !extPrebidServer.Empty() {
    requestExt.Prebid.Server = &serverInfo
}
```
Or should the `Empty` method go on type `Server` defined in `config/config.go` instead?

Reply:
Oh yes, I think I messed this up! I think you're right in saying we should only set `requestExt.Prebid.Server` when `e.server` is non-empty. And I could add an `Empty` method to check if `e.server` is non-empty.